Code/Tools/SerializeContextTools/Dumper.cpp

```diff
@@ -142,6 +149,333 @@ namespace AZ::SerializeContextTools
+        AZ::IO::SystemFile systemFile;
+        AZStd::variant<AZ::IO::StdoutStream, AZ::IO::SystemFileStream> outputStream;
+
+        AZ::CommandLine& commandLine = *application.GetAzCommandLine();
+        // If the output-file parameter has been supplied open the file path using FileIOStream
+        if (size_t optionCount = commandLine.GetNumSwitchValues("output-file"); optionCount > 0)
+        {
+            AZ::IO::FixedMaxPath outputPath;
+            if (AZ::IO::PathView outputPathView(commandLine.GetSwitchValue("output-file", optionCount - 1));
+                outputPathView.IsRelative())
+            {
+                AZ::Utils::ConvertToAbsolutePath(outputPath, outputPathView.Native());
+            }
+            else
+            {
+                outputPath = outputPathView.LexicallyNormal();
+            }
+
+            constexpr AZ::IO::OpenMode openMode = AZ::IO::OpenMode::ModeWrite | AZ::IO::OpenMode::ModeCreatePath;
+            auto& fileStream = outputStream.emplace<AZ::IO::SystemFileStream>(&systemFile, true);
+            if (!fileStream.Open(outputPath.c_str(), openMode))
+            {
+                AZ_Printf(
+                    "createtype",
+                    R"(Unable to specified output-file "%s". Object will not be dumped)"
+                    "\n",
+                    outputPath.c_str());
+                return false;
+            }
+        }
+
+        size_t typeIdOptionCount = commandLine.GetNumSwitchValues("type-id");
+        size_t typeNameOptionCount = commandLine.GetNumSwitchValues("type-name");
+        if (typeIdOptionCount == 0 && typeNameOptionCount == 0)
+        {
+            AZ_Error("createtype", false, "One of the following options must be supplied: --type-id or --type-name");
+            return false;
+        }
+        if (typeIdOptionCount > 0 && typeNameOptionCount > 0)
+        {
+            AZ_Error("createtype", false, "The --type-id and --type-name options are mutally exclusive. Only one can be specified");
+            return false;
+        }
+
+        AZ::SerializeContext* context = application.GetSerializeContext();
+        const AZ::SerializeContext::ClassData* classData = nullptr;
+        if (typeIdOptionCount > 0)
+        {
+            AZStd::string typeIdValue = commandLine.GetSwitchValue("type-id", typeIdOptionCount - 1);
+            classData = context->FindClassData(AZ::TypeId(typeIdValue.c_str(), typeIdValue.size()));
+            if (classData == nullptr)
+            {
+                AZ_Error("createtype", false, "Type with ID %s is not registered with the SerializeContext", typeIdValue.c_str());
+                return false;
+            }
+        }
+        else
+        {
+            AZStd::string typeNameValue = commandLine.GetSwitchValue("type-name", typeNameOptionCount - 1);
+            AZStd::vector<AZ::TypeId> classIds = context->FindClassId(AZ::Crc32{ AZStd::string_view{ typeNameValue } });
```

Comment:
Suggestion: As a future improvement, it would be pretty cool to extend this to allow partial matches, especially for the names where it can get a bit tricky specifying the namespaces too.

(On the other hand, after looking at the context code, handling partial matches would probably be a lot of work for very little payback.)

Reply:
Partial matches would be nice.
In order to finish the creation it would have to be narrowed down to one.

An alternative is regex matches as well where someone could do `(.+::)?:vector`

As long as it matched only a single entry, the type could be created